src/ompl/geometric/planners/bitstar/BITstar.h

```diff
@@ -0,0 +1,614 @@
+            problems. BIT* accomplishes this by processing batches of samples with a heuristic.
+            In doing so, it strikes a balance between algorithms like RRT* and FMT*.
+
+            @par J D. Gammell, S. S. Srinivasa, T. D. Barfoot, "Batch Informed Trees (BIT*): Sampling-based Optimal Planning via the Heuristically Guided Search of Implicit Random Geometric Graphs,"
+            To appear at ICRA 2015. <a href="http://arxiv.org/abs/1405.5848">arXiv:1405.5848 [cs.RO]</a>.
+            <a href="http://www.youtube.com/watch?v=MRzSfLpNBmA">Illustration video</a>.
+
+            @par TODO:
+            - Make k-nearest correct.
+            - Extend beyond single goal states to other samplable goals (i.e., goal sets).
+            - Generalize heuristics to make proper use of the optimization class.
+        */
+        /** \brief Batch Informed Trees (BIT*)*/
+        class BITstar : public ompl::base::Planner
+        {
+        public:
+            //Forward declarations so that the classes belong to BIT*:
+            /** \brief The vertex of implicit and explicit graphs. */
+            class Vertex;
+            /** \brief A generator of unique vertex IDs. */
+            class IdGenerator;
+            /** \brief The queue of edges to process as an integrated dual-stage queue (tracks both the expansion of vertices and the resulting edges) */
+            class IntegratedQueue;
+            //Helpful typedefs:
+            /** \brief A vertex shared pointer. */
+            typedef boost::shared_ptr<Vertex> VertexPtr;
+            /** \brief A \e constant vertex shared pointer. */
+            typedef boost::shared_ptr<const Vertex> VertexConstPtr;
+            /** \brief A vertex weak pointer. */
+            typedef boost::weak_ptr<Vertex> VertexWeakPtr;
+            /** \brief An integrated queue shared pointer. */
+            typedef boost::shared_ptr<IntegratedQueue> IntegratedQueuePtr;
+            /** \brief The vertex id type */
+            typedef unsigned int vid_t;
+
+            /** \brief Construct! */
+            BITstar(const base::SpaceInformationPtr& si, const std::string& name = "BITstar");
+
+            /** \brief Destruct! */
+            virtual ~BITstar();
+
+            /** \brief Setup */
+            virtual void setup();
+
+            /** \brief Clear */
+            virtual void clear();
+
+            /** \brief Solve */
+            base::PlannerStatus solve(const base::PlannerTerminationCondition& ptc);
+
+            /** \brief Get results */
+            virtual void getPlannerData(base::PlannerData& data) const;
+
+            /** \brief Get the next edge to be processed. Causes vertices in the queue to be expanded (if necessary) and therefore effects the run timings of the algorithm, but helpful for some videos and debugging. */
+            std::pair<ompl::base::State const*, ompl::base::State const*> getNextEdgeInQueue();
```

Comment:
Yes, it needs to be public, it's a function for making visualizations (and debugging).

I personally find `type const*` clearer, as when read right-to-left, you get "pointer to const type", and `type* const` gives "const pointer to type".

Will change tho'.